### PR TITLE
Support latest versions and consul agent

### PR DESCRIPTION
**Pull request by geek**

### CODE REVIEW

File: etc/containerpilot.json [hunk 1]
```diff
@@ -1,16 +1,53 @@
...
+      }
+    },
+    {
+      "name": "consul-agent",
+      "exec": ["/usr/local/bin/consul", "agent",
+                  "-data-dir=/data",
+                  "-config-dir=/config",
+                  "-log-level=err",
+                  "-rejoin",
+                  "-retry-join", "{{ .CONSUL | default "consul" }}",
```
Comment (tgross) [inline etc/containerpilot.json:43]: This assumes that we always have a Consul agent inside the container, which we need on Triton certainly but not when users reuse this blueprint in non-Triton environments.

File: Dockerfile [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
 # Alpine so we can use `docker build`.
 
 RUN apk add --update curl
 
 # add Prometheus. alas, the Prometheus developers provide no checksum
-RUN export prom=prometheus-1.3.0.linux-amd64 \
-    && curl -Lso /tmp/${prom}.tar.gz https://github.com/prometheus/prometheus/releases/download/v1.3.0/${prom}.tar.gz \
+RUN export PROM_VERSION=1.7.1 \
+    && export prom=prometheus-${PROM_VERSION}.linux-amd64 \
+    && curl -Lso /tmp/${prom}.tar.gz https://github.com/prometheus/prometheus/releases/download/v${PROM_VERSION}/${prom}.tar.gz \
```
Comment (jwreagor) [inline Dockerfile]: I'd probably download this archive and manually create a `sha256sum` checksum just to be thorough. You can follow the same pattern as Consul below this.

File: etc/containerpilot.json (hunk 1 above)
Comment (jwreagor) [reply etc/containerpilot.json:43]: Now I get it. I've been wondering why you (@tgross) place a conditional around the consul agent.

**Review by jwreagor — COMMENTED** [review]
Just a few comments, nothing earth shattering.

**Comment by tgross** [comment]
LGTM.

**Comment by geek** [comment]
@cheapRoc good to go, also checked consul-template, we weren't before

**Comment by misterbisson** [comment]
This, sadly, created a bunch of conflicts for https://github.com/autopilotpattern/prometheus/pull/18

**Pull request merged**